### ISSUE TITLE
How to get the curve in the paper?

### ISSUE THREAD

**Issue by eljrte** [lzhxmu/VTW#9]
<img width="739" height="776" alt="Image" src="https://github.com/user-attachments/assets/fb5832ec-e87e-43c8-84cc-8aee0240cd40" />

Hello, recently  I am studying your paper "Boosting Multimodal Large Language Models with Visual Tokens Withdrawal for Rapid Inference", and really interested in this picture, that the visiual token receives less attn in deep layer. Can u provide scripts or code to help me collect the data?Thansk very much.

**Comment by eljrte** [comment]
I tried to collect result on the Qwen2.5_vl model, but I got strange answer about the visual tokens : the deeper the layer ,the attn score began to rise. I wanna know if it stems from different models (as u use llava) or my code.
Below is part of my code, really appreciate u work：
`attn_prob = torch.softmax(attn_logits, dim=-1)
with torch.no_grad():
        B, H, Q, K = attn_prob.shape
        sample_heads = H
        sample_query = Q

        attn_weights_cpu = attn_prob[0, :sample_heads,:sample_query, 15+image_token_num:].float()

        part_score = attn_weights_cpu.mean(dim=0).sum().item()

        full_score_sum = attn_prob[0, :sample_heads, :sample_query, :].float().mean(dim=0).sum().item()
        
        partial_score_ratio = part_score / full_score_sum

        global layer_attn_cum
        layer_attn_cum[layer_idx] += partial_score_ratio
`
